prow/scripts/cluster-integration/kyma-cli-alpha-uninstall.sh

```diff
@@ -0,0 +1,192 @@
+trap gardener::cleanup EXIT INT
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c4)
+readonly COMMON_NAME_PREFIX="grdnr"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# Local variables
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+
+log::info "Building Kyma CLI"
+cd "${KYMA_PROJECT_DIR}/cli"
+make build-linux
+mv "${KYMA_PROJECT_DIR}/cli/bin/kyma-linux" "${KYMA_PROJECT_DIR}/cli/bin/kyma"
+export PATH="${KYMA_PROJECT_DIR}/cli/bin:${PATH}"
+
+log::info "Provision cluster: \"${CLUSTER_NAME}\""
+
+if [ -z "$MACHINE_TYPE" ]; then
+      export MACHINE_TYPE="n1-standard-4"
+fi
+
+gardener::provision_cluster
+
+log::info "Installing Kyma"
+
+# Parallel-install library installs cluster-essentials, istio, and xip-patch before kyma installation. That's why they should not exist on the InstallationCR.
+# Once we figure out a way to fix this, this custom CR can be deleted from this script.
+cat << EOF > "/tmp/kyma-parallel-install-installationCR.yaml"
+apiVersion: "installer.kyma-project.io/v1alpha1"
+kind: Installation
+metadata:
+  name: kyma-installation
+  namespace: default
+spec:
+  components:
+    - name: "testing"
+      namespace: "kyma-system"
+    - name: "knative-eventing"
+      namespace: "knative-eventing"
+    - name: "dex"
+      namespace: "kyma-system"
+    - name: "ory"
+      namespace: "kyma-system"
+    - name: "api-gateway"
+      namespace: "kyma-system"
+    - name: "rafter"
+      namespace: "kyma-system"
+    - name: "service-catalog"
+      namespace: "kyma-system"
+    - name: "service-catalog-addons"
+      namespace: "kyma-system"
+    - name: "helm-broker"
+      namespace: "kyma-system"
+    - name: "nats-streaming"
+      namespace: "natss"
+    - name: "core"
+      namespace: "kyma-system"
+    - name: "cluster-users"
+      namespace: "kyma-system"
+    - name: "logging"
+      namespace: "kyma-system"
+    - name: "permission-controller"
+      namespace: "kyma-system"
+    - name: "apiserver-proxy"
+      namespace: "kyma-system"
+    - name: "iam-kubeconfig-service"
+      namespace: "kyma-system"
+    - name: "serverless"
+      namespace: "kyma-system"
+    - name: "knative-provisioner-natss"
+      namespace: "knative-eventing"
+    - name: "event-sources"
+      namespace: "kyma-system"
+    - name: "application-connector"
+      namespace: "kyma-integration"
+    - name: "tracing"
+      namespace: "kyma-system"
+    - name: "monitoring"
+      namespace: "kyma-system"
+    - name: "kiali"
+      namespace: "kyma-system"
+    - name: "console"
+      namespace: "kyma-system"
+EOF
+
+(
+cd "${KYMA_PROJECT_DIR}/kyma"
+set -x
+kyma alpha deploy \
+    --ci \
+    --resources "${KYMA_PROJECT_DIR}/kyma/resources" \
+    --components "/tmp/kyma-parallel-install-installationCR.yaml"
+)
```

Comment:
This command is used 2 times and basically is similar to the one in the `gardener::install_kyma` function in the `gcp.sh` library. Will the `deploy` argument be the one that is used in the CLI or we will stay with `install`?
Additionally this part of code is the same as in the L162-170. Can it be moved to the separate function like `install_kyma`?

Reply:
The `deploy` is a new command that is being tested here, afaik it is install&upgrade. I could move it to the library, but as it is in the alpha state I was reluctant to do so.